burn-dataset/src/source/huggingface/mnist.rs

```diff
@@ -1,15 +1,58 @@
 use crate::source::huggingface::downloader::HuggingfaceDatasetLoader;
-use crate::{Dataset, InMemDataset};
+use crate::transform::{Mapper, MapperDataset};
+use crate::{Dataset, SqliteDataset};
+
+use image;
 use serde::{Deserialize, Serialize};
 
+const WIDTH: usize = 28;
+const HEIGHT: usize = 28;
+
 #[derive(Deserialize, Serialize, Debug, Clone)]
 pub struct MNISTItem {
-    pub image: [[f32; 28]; 28],
+    pub image: [[f32; WIDTH]; HEIGHT],
```

Comment:
Do you think it makes more sense to have the Image struct from the `image` crate instead of arrays?

Reply:
Great question! If you're thinking about a struct that just holds image data and can be serialized, then yeah, maybe. But if you mean adding methods and APIs to the Image, then I'd say no, and here's why:

- I'm treating Item as something that goes in and out of pre-processing. This means it's ready to go for computations and doesn't need extra work. I'm all about keeping the pipeline neat and tidy, so each node takes in data and gives out data. Any tweaks to the image or audio should be done on the raw bytes. That way, the next step in the processing pipeline can do its thing without a hitch.
  
- I want to keep Item serializable because sometimes you need to stash it on your local disk after a bunch of preprocessing. I'm working on a pre-processor trait to make this more general.

- Lastly, the folks using the dataset for training or testing shouldn't have to worry about what APIs to call or how to turn it into usable data. For example, they shouldn't need to know to call `to_pixel()`.